### PR TITLE
i#3315,i#3348: Move mem{cpy,set,move} into memfuncs on AArchXX

### PR DESCRIPTION
**Pull request by derekbruening**
Provides isolation of mem{cpy,set,move} in the DR static library on
AArchXX.  Moves the assembly mem{cpy,set} into a separate memfuncs.asm
for arm and aarch64, and links in memmove.c.  Marks the
__mem{cpy,set}_chk aliases as weak.  This brings AArchXX into parity
with x86 on library symbol isolation.

Removes the exceptions for AArchXX in the CMake_symbol_check test.
Confirms the build-time test fails without the code changes here:

  CMake Error at xxx/core/CMake_symbol_check.cmake:98 (message):
    *** Error:
    xxx/build_a64_dbg_tests/lib64/debug/libdynamorio_static_nohide.a
    contains a likely-to-conflict symbol: 4279: 00000000002cea3c 0 FUNC GLOBAL
    HIDDEN 248 memcpy

Issue: #3315, #3348

### CODE REVIEW

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -80,13 +80,13 @@ endif ()
 add_asm_target(arch/pre_inject_asm.asm preinject_asm_src preinject_asm_tgt ""
   "-DNOT_DYNAMORIO_CORE_PROPER" "${asm_deps}")
 
-if (UNIX AND X86)
```
Comment (xyzsam) [inline core/CMakeLists.txt:83]: Just a reminder that there is an outstanding bug on the X86 macro definition causing conflicts in google3.

**Comment by derekbruening** [comment]
> I think you're missing memfuncs.asm for arm in this commit - I only see deletions.

Wow, thank you -- never added locally.  Hmm, now I'm wondering how the AArch64 tests passed -- ok and now that I added them we have them failing :(.  Local AArch64 build was good.  Investigating.

**Comment by derekbruening** [comment]
The A64 failure here is strange: I am theorizing a build race due a missing dependence on api_headers.  And I guess all the x86 builds just never hit it.

For A64 succeeding after I removed mem* and didn't add it back: there is a check that libdynamorio has zero libc imports that I would expect to fail with the compiler substituting libc versions.  Maybe it inlined everything?

**Comment by derekbruening** [comment]
> For A64 succeeding after I removed mem* and didn't add it back: there is a check that libdynamorio has zero libc imports that I would expect to fail with the compiler substituting libc versions. Maybe it inlined everything?

Yeah I guess it would fail first due to not finding the referenced asm file :)  And looking at the output:

http://139.178.83.194:8080/job/DynamoRIO-AArch64-Precommit/1382/console

```
CMake Error: Cannot determine link language for target "drmemfuncs".
CMake Error in core/CMakeLists.txt:
  Exporting the target "drmemfuncs" is not allowed since its linker language
  cannot be determined
Command exited with the value: 1
Error(s) when configuring the project
```

So it did indeed fail.  But the wrapper script fails to detect the config-time failure as an overall error (too loose in ignoring cross-arch build attempts in multi-build setup?) and it showed up green.  Not good.  It seems familiar: maybe I'm thinking of Mac #4022?

**Comment by derekbruening** [comment]
I filed #4312 on the misleading green on config failure.  And now AArch64 is green and I double-checked and it's a true green.  So I think we're all set here for this PR.
PTAL.

**Comment by xyzsam** [comment]
Looks like appveyor is still failing?

**Comment by derekbruening** [comment]
> Looks like appveyor is still failing?

Looks like a flake that has started becoming more common: drwrap-test-detach under #4058.  Should be added to the flake list since probably nobody will have time to try to fix it (if someone can reproduce it: most of these Appveyor flakes I can't repro on my recent-Win10 machine: Appveyor I believe is still on Win10-1607).  I clicked "re-run" on appveyor just to see how flaky this is.  Looks like @johnfxgalea hit it on a recent PR as well.

**Comment by derekbruening** [comment]
I have PR #4313 marking it as a flake.

**Comment by derekbruening** [comment]
And PR #4314 fixes the false green.

**Comment by johnfxgalea** [comment]
> Looks like @johnfxgalea hit it on a recent PR as well.

**Pull request merged**